pyingest/parsers/dubcore.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+import sys
+import json
+import codecs
+from default import BaseXmlToDictParser
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+
+class DublinCoreParser(BaseXmlToDictParser):
+
+    def __init__(self):
+    # make sure we are utf-8 clean on stdout, stderr
+        self.DUBC_SCHEMA = "http://www.openarchives.org/OAI/2.0/oai_dc/"
+
+
+    def get_schema(self, r):
+        schema = []
+        for s in self._array(r[u'@xmlns:oai_dc']):
+            schema.append(self._text(s))
+        if len(schema) == 0:
+            raise NoSchemaException("Unknown record schema.")
+            schema = None
+        elif schema[0] != self.DUBC_SCHEMA:
+            raise WrongSchemaException("Wrong schema.")
+        else:
+            pass
+        schema = ['dc:contributor','dc:coverage','dc:creator','dc:date',
+                  'dc:description','dc:format','dc:identifier','dc:language',
+                  'dc:publisher','dc:relation','dc:rights','dc:source',
+                  'dc:subject','dc:title','dc:type']
+        return schema
+
+
+    def get_tag(self, r, tag):
+        value = []
+        try:
+            tag,self._array(r.get(tag,[]))
```

Comment:
Don't quite understand why this is necessary (calling the self._array function twice, once inside of try and then outside).
Also r.get(key,default) should never generate a KeyError AFAIK